tiny_cnn/lossfunctions/loss_function.h

```diff
@@ -206,6 +206,7 @@ void apply_cost_if_defined(std::vector<vec_t>& sample_gradient, const std::vecto
         }
     }
 }
+constexpr void *supress_unused_warning_apply_cost_if_defined = (void*)(apply_cost_if_defined);
```

Comment:
try this
    
      constexpr void* supress_unused_warning_apply_cost_if_defined =
            reinterpret_cast<void*>(&apply_cost_if_defined);